editor/src/components/inspector/sections/component-section/data-picker-utils.tsx

```diff
@@ -1,50 +1,48 @@
 import { atom } from 'jotai'
+import { processJSPropertyAccessors } from '../../../../core/data-tracing/data-tracing'
+import { MetadataUtils } from '../../../../core/model/element-metadata-utils'
+import { foldEither } from '../../../../core/shared/either'
+import * as EP from '../../../../core/shared/element-path'
+import type { ElementPathTrees } from '../../../../core/shared/element-path-tree'
+import type { ElementInstanceMetadataMap } from '../../../../core/shared/element-template'
 import {
   isJSExpression,
   type JSExpressionOtherJavaScript,
   type JSXElementChild,
 } from '../../../../core/shared/element-template'
+import type { ElementPath } from '../../../../core/shared/project-file-types'
 import { assertNever } from '../../../../core/shared/utils'
+import type { AllElementProps } from '../../../editor/store/editor-state'
 import type { ArrayInfo, JSXInfo, ObjectInfo, PrimitiveInfo } from './variables-in-scope-utils'
-import { processJSPropertyAccessors } from '../../../../core/data-tracing/data-tracing'
-import { foldEither } from '../../../../core/shared/either'
 
-export interface PrimitiveOption {
-  type: 'primitive'
-  variableInfo: PrimitiveInfo
-  definedElsewhere: string
+interface VariableOptionBase {
```

Comment:
VariableOptionBase is a new common base type all VariableOptions extend, this makes the code a little bit easier to read